File: ChipsGame/JavaVersion/JackPlayer.java
/**
 * This is the class that will be used in the contest, it is about even
 * with a class that is similar to itself, but it beats me most of the time
 * 
 * @author Jack Hughes
 * @version 03/31/17
 */
public class JackPlayer implements Player {

	private String name = "Jack";

	/**
	 * Default constructor for the JackPlayer, sets name to Jack
	 */
	public JackPlayer(){
		name = "Jack";
	}
	
	/**
	 * Sets JackPlayer to a different name besides Jack
	 */
	public JackPlayer(String namer){
		name = namer;
	}
	
	/**
	 * This method is smarter than the other Players we have made so far
	 * but is right about even with a class that is slightly different, 
	 * it likes to start with positive numbers better 
	 */
	@Override
	public int getMove(Game g) {
		if(g.getLimitMove() >= g.getPile()){
			return g.getPile();
		}
		else if(g.getPile() == g.getLimitMove() + 1){
			return 2;
		}
		else if(g.getPile() <= 100 && g.getPile()%2==0){
			return 1;
		}
		else return 2;
	}

	/**
	 * Returns the name Jack
	 */
	@Override
	public String getName() {
		return name;
	}
}

File: ChipsGame/PythonVersion/jackPlayer.py
'''
Created on Sep 29, 2017

This class is based around the idea that you never want to be down to 3 chips and
can only take away 2, if the pile is even it takes away 1 because it is trying to
get the other player to have 3 by the end, if it is odd it will take away 2
because again it is trying to make the other player have an odd number at the end
so that it will win 

@author: jack
'''

class JackPlayer:
    
    #sets the name for the class
    def __init__(self, name):
        self.name = name        
    
    """takes chips away from the pile based on if the pile is even
    or odd basically
    """    
    def getMove(self, g):
        self.Game = g
        if self.Game.getLimitMove() >= self.Game.getPile():
            return self.Game.getPile()
        elif self.Game.getPile() == self.Game.getLimitMove() + 1:
            return 2
        #elif y <= 100 and y%2==0:
        elif self.Game.getPile() <= 100 and self.Game.getPile()%2==0:
            return 1
        else:
            return 2
        
    #returns the set name    
    def getName(self):
            return self.name
    
    
    